src/util/future.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::boxed;
+use futures::sync::oneshot;
+
+pub fn future_from_callback<T>() -> (Box<boxed::FnBox(T) + Send>, oneshot::Receiver<T>)
```

Comment:
I think this function can move to futurepool.rs.

Reply:
Similar functionality is used elsewhere, mostly named as `make_callback`. I extracted it into the util for reuse.

Reply:
`future_from_callback` sounds like construct future from callback, but actually it is not.

Reply:
Its something like "tie them together". Suggestions?

Reply:
gen_cb_future_pair